VK/VK/AuthorizationScreen/AuthorizationViewController.swift

```diff
@@ -0,0 +1,102 @@
+// AuthorizationViewController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// Экран авторизации.
+final class AuthorizationViewController: UIViewController {
+    // MARK: - private visual components
+
+    @IBOutlet private var authorizationScrollView: UIScrollView!
+    @IBOutlet private var loginWithAppleButton: UIButton!
+    @IBOutlet var loginTextField: UITextField!
+
+    // MARK: - life cycle
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        addObserves()
+        configLoginWithAppleButton()
+        addTapGestoreRecognizer()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        removeObsevers()
+    }
+
+    override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
+        if identifier == "loginSegue" {
+            guard let loginTextFieldText = loginTextField.text else { return false }
+            guard loginTextFieldText == "@gai1_grigorenko" else { return false }
+            return true
+        } else {
+            return false
+        }
+    }
+
+    // MARK: - private methods
+
+    private func addTapGestoreRecognizer() {
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(hideKeyboard))
+        authorizationScrollView.addGestureRecognizer(tapGesture)
+    }
+
+    private func addObserves() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillShown),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillShown),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    private func removeObsevers() {
+        NotificationCenter.default.removeObserver(
+            self,
+            name: UIResponder.keyboardDidHideNotification,
+            object: nil
+        )
+
+        NotificationCenter.default.removeObserver(
+            self,
+            name: UIResponder.keyboardDidShowNotification,
+            object: nil
+        )
+    }
+
+    private func configLoginWithAppleButton() {
+        loginWithAppleButton.layer.borderWidth = 1
+        loginWithAppleButton.layer.borderColor = UIColor.black.cgColor
+    }
+
+    @objc private func keyboardWillShown(notification: Notification) {
```

Comment:
```suggestion
    @objc private func keyboardWillShown(notification: Notification) {
```
все @objc private func Должны быть выше Private Methods
Пример:
MARK: Button Action 
@objc private func buttonAction
......
MARK: Private Methods
private func someFunc 
.....